test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -0,0 +1,156 @@
+			prodWl := testing.MakeWorkload("prod-wl", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl)).Should(gomega.Succeed())
+			devWl := testing.MakeWorkload("dev-wl", ns.Name).Queue(devQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, devWl)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl)
+			framework.ExpectWorkloadsToBeWaiting(ctx, k8sClient, devWl)
+
+			ginkgo.By("update the first workload to be in the PodsReady condition and verify the second workload is admitted")
+			prodKey := types.NamespacedName{Name: prodWl.Name, Namespace: prodWl.Namespace}
+			gomega.Expect(k8sClient.Get(ctx, prodKey, prodWl)).Should(gomega.Succeed())
+			apimeta.SetStatusCondition(&prodWl.Status.Conditions, metav1.Condition{
+				Type:   kueue.WorkloadPodsReady,
+				Status: metav1.ConditionTrue,
+				Reason: "PodsReady",
+			})
+			gomega.Expect(k8sClient.Status().Update(ctx, prodWl)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, devClusterQ.Name, devWl)
+		})
+
+		ginkgo.It("Should unblock admission of new workloads once the admitted workload is deleted", func() {
+			ginkgo.By("checking the first workload gets admitted while the second is waiting")
+			prodWl := testing.MakeWorkload("prod-wl", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl)).Should(gomega.Succeed())
+			devWl := testing.MakeWorkload("dev-wl", ns.Name).Queue(devQueue.Name).Request(corev1.ResourceCPU, "2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, devWl)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, prodClusterQ.Name, prodWl)
+			framework.ExpectWorkloadsToBeWaiting(ctx, k8sClient, devWl)
+
+			ginkgo.By("delete the first workload and verify the second workload is admitted")
+			gomega.Expect(k8sClient.Delete(ctx, prodWl)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, devClusterQ.Name, devWl)
+		})
+
+		ginkgo.It("Should block admission of one new workload if two are considered in the same scheduling cycle", func() {
+			ginkgo.By("creating two workloads but delaying cluster queue creation which has enough capacity")
+			prodWl := testing.MakeWorkload("prod-wl", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "11").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl)).Should(gomega.Succeed())
+			devWl := testing.MakeWorkload("dev-wl", ns.Name).Queue(devQueue.Name).Request(corev1.ResourceCPU, "11").Obj()
+			gomega.Expect(k8sClient.Create(ctx, devWl)).Should(gomega.Succeed())
+			framework.ExpectWorkloadsToBePending(ctx, k8sClient, prodWl, devWl)
+
+			ginkgo.By("creating the cluster queue to ensure the two workloads are considered in the same scheduling cycle")
```

Comment:
The rest is already explained in the first `By` and in the comment

```suggestion
			ginkgo.By("creating the cluster queue")
```